### ISSUE TITLE
Keyboard navigation misbehaves after clicking checkbox in multi-selection column

### ISSUE THREAD

**Issue by Haprog** [vaadin/vaadin-grid#1732]
### Description
With a multi-select grid, when you click the row selection checkbox with the mouse, the checkbox component itself receives focus and in that case using arrow keys scrolls the scrollable area of the grid directly instead of navigating between the cells/rows. This only happens with the combination of mouse + keyboard.

If you only navigate the grid with the keyboard, you can also navigate to the checkbox cell with the arrow keys and check/uncheck it with the spacebar key, but in that case the focus is on the whole cell that contains the checkbox (not in the checkbox itself).

Related forum thread: https://vaadin.com/forum/thread/18252091/18252092

### Expected outcome
After clicking the selection checkbox with the mouse, the cell should be focused so that keyboard navigation works as expected after mouse click.

### Actual outcome
When pressing arrow up/down after clicking a checkbox, the grid scrolls instead of navigating between rows.

### Live Demo
See Grid Multi Selection demo at https://vaadin.com/components/vaadin-grid/java-examples/selection

### Steps to reproduce
1. Click on any of the checkboxes in body rows.
2. Try to navigate with arrow keys up/down.

### Browsers Affected
At least Chrome 81 and Firefox 75.

**Comment by tomivirkki** [comment]
The behavior was originally designed so that clicking any focusable element inside a cell wouldn't result in its native focus being stolen to the grid cell. For example, if you have an `<input>` inside a cell, you should be able to focus it by clicking and then move the caret inside of it with arrow keys, or you could focus another `<input>` inside the same cell by hitting Tab. This is the so-called "interaction mode".

You can exit the "interaction mode" (and enter the "navigation mode") by hitting Esc. Respectively, you can enter interaction mode again by hitting Enter on a focused cell (this will focus the first focusable element inside the cell). Rather than changing this behavior, we should prevent the grid from being scrolled with arrow keys when in "interaction mode".

**Comment by Haprog** [comment]
Thanks for the explanation but I still think `<vaadin-grid-selection-column>` should prevent entering the "interaction mode" when clicking on the checkboxes.

> Rather than changing this behavior, we should prevent the grid from being scrolled with arrow keys when in "interaction mode".

I'm not sure about this. Might make sense for "normal cells" that may contain custom interactive elements, but I'm not sure if it would be more confusing to prevent scroll in case you are in interaction mode but the focused element doesn't do anything on arrow up/down. Maybe it's better to leave it as is for generic cases.

This issue is only about the `<vaadin-grid-selection-column>` where I don't think "interaction mode" gives any benefit.

**Comment by rolfsmeds** [comment]
I agree that the selection checkbox is a special case -- the only potential downside I see with changing it not to enter interaction mode is that behavior would be inconsistent with regards to checkboxes used in component renderers, but I see that as a smaller problem.

Ideally it would be great if any component embedded in body cells could somehow be configured to not trigger interaction mode.
